catalog/core/catalog-core-contentresourcereader/src/main/java/org/codice/ddf/catalog/content/resource/reader/ContentResourceReader.java

```diff
@@ -140,6 +142,14 @@ public ResourceResponse retrieveResource(URI resourceUri, Map<String, Serializab
             LOGGER.debug("Resource URI is content scheme");
             String contentId = resourceUri.getSchemeSpecificPart();
             if (contentId != null && !contentId.isEmpty()) {
+                if (arguments != null && StringUtils.isNotBlank((String)arguments.get("options"))){
```

Comment:
Are you guaranteed that get("options") always returns a String? Should you add a check to make sure it is String to avoid a class cast exception?